src/adcm_pytest_plugin/docker_utils.py

```diff
@@ -283,16 +284,31 @@ def _wait_for_adcm_container_init(container, container_ip, port, timeout=120):
         raise TimeoutError(f"ADCM API has not responded in {timeout} seconds{additional_message}")
 
 
+@dataclass
+class ContainerConfig:
+    """Dataclass for incapsulating docker container run options"""
+
+    image: str = "hub.arenadata.io/adcm/adcm"
+    tag: str = "latest"
+    pull: bool = True
+    ip: str = DEFAULT_IP
+    labels: dict = field(default_factory=dict)
+    remove: bool = True
+    volumes: Optional[dict] = None
+    name: Optional[str] = None
+
+
 class ADCM:
     """
     Class that wraps ADCM Api operation over self.api (ADCMApiWrapper)
     and wraps docker over self.container (see docker module for info)
     """
 
-    __slots__ = ("container", "ip", "port", "url", "api")
+    __slots__ = ("container", "config", "ip", "port", "url", "api")
 
-    def __init__(self, container, ip, port):
+    def __init__(self, container: Container, ip: str, port: str, config: ContainerConfig):
```

Comment:
Hereinafter.
It may be worth using the explicit `container_config` variable name. In the current version, when `config` is used in different contexts, it is misleading

Reply:
Changed in `ADCM` only because it is the only class which will be exposed as fixture value.
Other usages of `config` have no another context.